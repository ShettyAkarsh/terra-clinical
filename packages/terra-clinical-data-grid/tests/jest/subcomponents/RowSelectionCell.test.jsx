import React from 'react';
import RowSelectionCell from '../../../src/subcomponents/RowSelectionCell';

describe('Cell Snapshot Tests', () => {
  it('should render a Cell with minimal props', () => {
    const cell = shallow((
      <RowSelectionCell
        sectionId="section-test"
        rowId="row-test"
        columnId="column-test"
        width="123px"
      />
    ));

    expect(cell).toMatchSnapshot();
  });

  it('should render a Cell with all provided props', () => {
    const cell = shallow((
      <RowSelectionCell
        sectionId="section-test"
        rowId="row-test"
        columnId="column-test"
        width="123px"
        isSelectable
        isSelected
        onSelect={() => {}}
        onHoverStart={() => {}}
        onHoverEnd={() => {}}
        selectableRefCallback={() => {}}
        ariaLabel="test label"
      />
    ));

    expect(cell).toMatchSnapshot();
  });
});
