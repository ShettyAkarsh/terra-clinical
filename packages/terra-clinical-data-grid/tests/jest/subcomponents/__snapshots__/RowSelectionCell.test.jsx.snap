// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell Snapshot Tests should render a Cell with all provided props 1`] = `
<Cell
  ariaLabel="test label"
  className="selectable selected"
  columnId="column-test"
  isSelectable={true}
  onHoverEnd={[Function]}
  onHoverStart={[Function]}
  onSelect={[Function]}
  rowId="row-test"
  sectionId="section-test"
  selectableRefCallback={[Function]}
  width="123px"
>
  <div
    className="icon-container"
  >
    <div
      className="icon"
    />
  </div>
</Cell>
`;

exports[`Cell Snapshot Tests should render a Cell with minimal props 1`] = `
<Cell
  aria-hidden={true}
  className=""
  columnId="column-test"
  onSelect={[Function]}
  rowId="row-test"
  sectionId="section-test"
  width="123px"
>
  <div
    className="icon-container"
  >
    <div
      className="icon"
    />
  </div>
</Cell>
`;
